xla/service/gpu/stream_attribute_annotator.h

```diff
@@ -45,6 +45,11 @@ namespace xla::gpu {
 
 class StreamAttributeAnnotator : public HloModulePass {
  public:
+  // copy_start_ is used to differentiate copy-start from
+  // other instructions as it doesn't need the annotations to split into
+  // AsyncStart and AsyncDone: copy-done is available already.
+  explicit StreamAttributeAnnotator(bool copy_start = false)
```

Comment:
Sorry I still don't follow. This is invoked from gpu_compiler.cc with copy_start set to `true`. Is it ever set to `false`? When do we invoke this with `true` and when do we invoke this with `false`?

Reply:
> This is invoked from gpu_compiler.cc with copy_start set to true. 
Yes, it needs to happen after HLO rematerialization pass in RunPostSchedulingPipelines() because rematerialization can create copy-start and StreamAttributeAnnotator adds annotation to copy-start when copy_start set to true.
> Is it ever set to false? When do we invoke this with true and when do we invoke this with false?
It is set to false by default in the existing code: https://github.com/openxla/xla/pull/10636/files#diff-fc02eb6aea06ad0d72011e9a64da28e8c6fae000a9e0cab9b15b40b4e914af4aL1201. It's called in OptimizeHloModule(). It's too early for us as rematerialization is not called yet and copy-start is not created.
So StreamAttributeAnnotator is called twice right now. This way, I don't have to change the original invocation. Including @Tixxx into the loop as he created this pass.